sdk/cosmosdb/cosmos/test/public/functional/NonStreamingQueryPolicy.spec.ts

```diff
@@ -692,4 +698,221 @@ describe.skip("Full text search feature", async () => {
+    const { resources } = await container.items.query(query).fetchAll();
+    // Should only return items from partition 'A' that match
+    assert.equal(resources.length, 1);
+    assert.equal(resources[0].pk, "A");
+    assert.equal(resources[0].text, "I like to swim");
+
+    await database.container(containerName).delete();
+  });
+
+  it("should execute a cross-partition full text query", async () => {
+    const containerName = "full text search cross partition test";
+    const { container } = await database.containers.createIfNotExists({
+      id: containerName,
+      partitionKey: { paths: ["/pk"] },
+      indexingPolicy: {
+        includedPaths: [{ path: "/*" }],
+        fullTextIndexes: [{ path: "/text" }],
+      },
+      fullTextPolicy: {
+        defaultLanguage: "en-US",
+        fullTextPaths: [{ path: "/text", language: "en-US" }],
+      },
+      throughput: 12000,
+    });
+
+    // Insert items into different partitions
+    await container.items.create({ id: "1", pk: "A", text: "I like to swim" });
+    await container.items.create({ id: "2", pk: "B", text: "I like to swim" });
+    await container.items.create({ id: "3", pk: "C", text: "I like to run" });
+
+    // Query across all partitions for 'swim'
+    const query = "SELECT * FROM c WHERE FullTextContains(c.text, 'swim')";
+    const { resources } = await container.items.query(query, { forceQueryPlan: true }).fetchAll();
+    // Should return items from all partitions that match
+    const pks = resources.map((r) => r.pk);
+    assert.equal(resources.length, 2);
+    assert(pks.includes("A"));
+    assert(pks.includes("B"));
+
+    await database.container(containerName).delete();
+  });
+
+  it("should execute a full text query using hierarchical partition keys", async () => {
+    const containerName = "full text search hierarchical partition test";
+    // Create a container with hierarchical partition keys and full text policy
+    const { container } = await database.containers.createIfNotExists({
+      id: containerName,
+      partitionKey: { paths: ["/country", "/city"], version: 2, kind: PartitionKeyKind.MultiHash },
+      indexingPolicy: {
+        includedPaths: [{ path: "/*" }],
+        fullTextIndexes: [{ path: "/text" }],
+      },
+      fullTextPolicy: {
+        defaultLanguage: "en-US",
+        fullTextPaths: [{ path: "/text", language: "en-US" }],
+      },
+      throughput: 12000,
+    });
+
+    // Insert items into different hierarchical partitions
+    await container.items.create({ id: "1", country: "US", city: "NY", text: "I like to swim" });
+    await container.items.create({ id: "2", country: "US", city: "SF", text: "I like to swim" });
+    await container.items.create({
+      id: "3",
+      country: "CA",
+      city: "Toronto",
+      text: "I like to run",
+    });
+
+    // Query only partition ['US', 'NY'] for 'swim'
+    const query = "SELECT * FROM c WHERE FullTextContains(c.text, 'swim')";
```

Comment:
Does gateway directly support `FullTextContains` queries? Configuring partition key in options only work if gateway is directly able to serve up the query. If we have to use our query pipeline, this valueis skipped. Probably a bug in our SDK.